sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusReceiverAsyncClient.java

```diff
@@ -268,47 +268,23 @@ public String getServiceBusResourceName() {
         return deadLetter(message, null);
     }
 
-    /**
-     * Moves a {@link ServiceBusMessage} to the deadletter sub-queue with modified message properties.
-     *
-     * @param message to be used.
-     * @param propertiesToModify Message properties to modify.
-     *
-     * @return The {@link Mono} the finishes this operation on service bus resource.
-     */
-    public Mono<Void> deadLetter(ServiceBusReceivedMessage message, Map<String, Object> propertiesToModify) {
-        return deadLetter(message, null, null, propertiesToModify);
-    }
-
-    /**
-     * Moves a {@link ServiceBusMessage} to the deadletter sub-queue with deadletter reason and error description.
-     *
-     * @param message to be used.
-     * @param deadLetterReason The deadletter reason.
-     * @param deadLetterErrorDescription The deadletter error description.
-     *
-     * @return The {@link Mono} the finishes this operation on service bus resource.
-     */
-    public Mono<Void> deadLetter(ServiceBusReceivedMessage message, String deadLetterReason,
-        String deadLetterErrorDescription) {
-        return deadLetter(message, deadLetterReason, deadLetterErrorDescription, null);
-    }
 
     /**
      * Moves a {@link ServiceBusMessage} to the deadletter sub-queue with deadletter reason, error description and
      * modifided properties.
      *
      * @param message to be used.
-     * @param deadLetterReason The deadletter reason.
-     * @param deadLetterErrorDescription The deadletter error description.
-     * @param propertiesToModify Message properties to modify.
+     * @param deadLetterOptions The options to specify while moving message to the deadletter sub-queue.
      *
      * @return The {@link Mono} the finishes this operation on service bus resource.
      */
-    public Mono<Void> deadLetter(ServiceBusReceivedMessage message, String deadLetterReason,
-        String deadLetterErrorDescription, Map<String, Object> propertiesToModify) {
-        return updateDisposition(message, DispositionStatus.SUSPENDED, deadLetterReason, deadLetterErrorDescription,
-            propertiesToModify);
+    public Mono<Void> deadLetter(ServiceBusReceivedMessage message, DeadLetterOptions deadLetterOptions) {
+        if (deadLetterOptions != null) {
```

Comment:
* It would be easier to create a static default_dead_letter_options and then pass the arguments in, so that we don't do this extra logic if it is null.

Reply:
Since this is an optional parameter, I expected another overload that just takes the received message and under the covers would pass default_dead_letter_options, that way we can do a null object assertion in this method.